adserver/views.py

```diff
@@ -650,19 +647,17 @@ def ignore_tracking_reason(self, request, advertisement, offer):
         elif not offer.publisher:
             log.log(self.log_level, "Ad impression for unknown publisher")
             reason = "Unknown publisher"
-        elif not advertisement.flight.show_to_geo(
-            country_code, region_code, metro_code
-        ):
+        elif not advertisement.flight.show_to_geo(geo_data):
             # This is very rare but it is visible in ad reports
             # I believe the most common cause for this is somebody uses a VPN and is served an ad
             # Then they turn off their VPN and click on the ad
             log.log(
```

Comment:
I've been enjoying structlog on the RTD side. Might be worth looking into, as it lets you `structlog.bind` a set of variables that carry over into all later logs in that request, which leads to some nice outcomes.

Reply:
I've never used it but it looks like it has some nice features. I have a branch I've been toying with that does some test only changes and that might be a good place to try out some logging changes.